component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/OpenIdProcessorImpl.java

```diff
@@ -261,88 +259,17 @@ protected OAuth2AccessToken parseResponse(String httpResponse) throws JSONExcept
-                    String errorMessage = "Error when obtaining content from OpenId. Error details: " + httpResponse.getResponse();
-                    log.warn(errorMessage);
-                    throw new OAuthException(OAuthExceptionCode.ACCESS_TOKEN_ERROR, errorMessage);
-                } else {
-                    String errorMessage = "Unspecified IO error. Http response code: " + httpResponse.getResponseCode() + ", details: " + httpResponse.getResponse();
-                    log.warn(errorMessage);
-                    throw new OAuthException(OAuthExceptionCode.IO_ERROR, errorMessage);
-                }
-            }
-
-
-            @Override
-            protected JSONObject parseResponse(String httpResponse) throws JSONException {
-                JSONObject json = new JSONObject(httpResponse);
-                return json;
-            }
-        }.executeRequest(params);
-
-        String[] scopes;
+        String scope = accessToken.getTokenData().getScope();
         try {
-            // If there was an error in the token info, abort.
-            if (tokenInfo.has("error")) {
-                throw new OAuthException(OAuthExceptionCode.ACCESS_TOKEN_ERROR,
-                        "Error during token validation: " + tokenInfo.get("error").toString());
-            }
-
-            //depending of openid implementation
-            //issued_to can be named client_id
-            String clientId = tokenInfo.has(OAuthConstants.ISSUED_TO_PARAMETER) ?
-                    tokenInfo.getString(OAuthConstants.ISSUED_TO_PARAMETER) :
-                    tokenInfo.has(OAuthConstants.CLIENT_ID_PARAMETER) ?
-                            tokenInfo.getString(OAuthConstants.CLIENT_ID_PARAMETER) : null;
-
-            if (clientId != null && !clientId.equals(clientID)) {
-                throw new OAuthException(OAuthExceptionCode.ACCESS_TOKEN_ERROR,
-                        "Token's client ID does not match app's. clientID "
-                                + "from tokenINFO: " + tokenInfo.get(OAuthConstants.ISSUED_TO_PARAMETER));
-            }
-
-            if (log.isTraceEnabled()) {
-                log.trace("Successfully validated accessToken from openid : " + tokenInfo);
-            }
-            scopes = tokenInfo.has(OAuthConstants.SCOPE_PARAMETER) ? tokenInfo.getString(OAuthConstants.SCOPE_PARAMETER).split(" ") : null;
-        } catch (JSONException jsonException) {
-            throw new OAuthException(OAuthExceptionCode.ACCESS_TOKEN_ERROR,
-                    "Error during token validation: token format is ko");
+            Jwts.parserBuilder()
+                    .setSigningKeyResolver(remoteJwkSigningKeyResolver)
+                    .requireIssuer(this.issuer)
+                    .requireAudience(this.clientID)
+                    .build();
+        } catch (MissingClaimException e) {
+            log.error("the parsed JWT did not have the sub field");
```

Comment:
What do you return in case of error ? we should not return the token as it worked.